sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetPartitionDiscoverySuite.scala

```diff
@@ -1067,4 +1098,24 @@ class ParquetPartitionDiscoverySuite extends QueryTest with ParquetTest with Sha
       checkAnswer(spark.read.load(path.getAbsolutePath), df)
     }
   }
+
+  test("Resolve type conflicts - decimals, dates and timestamps in partition column") {
+    withTempPath { path =>
+      val df = Seq((1, "2015-01-01"), (2, "2016-01-01 00:01:00")).toDF("i", "ts")
+      df.write.format("parquet").partitionBy("ts").save(path.getAbsolutePath)
```

Comment:
this passes even before the change, right?

as I mentioned on the other PR, I don't really understand why this works despite the issue you're fixing.  Regardless, seems like a good test to have.

Reply:
I think `2016-01-01 00:01:00` was placed when calling `resolveTypeConflicts`.  Let me try to make this test case not dependent on this.